### PR TITLE
Switch client to backend widgets

### PR DESCRIPTION
**Pull request by cbarber**
This PR is the first step towards overhauling the data on the client. It moves  the list of enabled and available (based on time of day) widgets  to a backend query. This opens up switching all widgets to follow a pattern of exporting `{ component, queryFragment, subscriptionFragment }` so the widget controller can use the fragments when retrieving the current node.

### CODE REVIEW

File: app/javascript/components/widget-controller.jsx [hunk 1]
```diff
@@ -1,66 +1,85 @@
...
       if (transitionTime < now) {
-        this.moveDown();
+        this.nextWidget();
+      } else if (
+        prefetchTime < now &&
+        this.lastPrefetchTime < now - prefetchDelay
+      ) {
+        const { nextWidgetId } = this.state;
+        this.lastPrefetchTime = now;
+        client.query({
```
Comment (chewnoill) [inline app/javascript/components/widget-controller.jsx]: do we have hooks here? hooks would be nice.
https://www.apollographql.com/docs/react/api/react-hooks/#uselazyquery

File: app/models/widget.rb [hunk 2]
```diff
@@ -10,7 +10,8 @@ class Widget < ApplicationRecord
   default_scope { order(position: :asc) }
   scope :enabled, -> { where(enabled: true) }
   scope :available, -> do
-    tod = Tod::TimeOfDay(Time.now).to_s
+    # TimeOfDay does not work with timezone
+    tod = Tod::TimeOfDay(Time.now).to_s # rubocop:disable Rails/TimeZone
```
[the remark this answers is not in the document]
Comment (cbarber) [reply app/models/widget.rb]: Oops. Looks like I might have done a fixup in the wrong commit.

File: app/javascript/components/widget-controller.jsx [hunk 3]
```diff
@@ -45,9 +45,32 @@ export class WidgetController extends React.Component {
...
+      const { transitionTime } = this.state;
+
+      if (!transitionTime) {
+        return;
+      }
+
+      const now = Date.now();
+
+      if (transitionTime < now) {
+        this.moveDown();
```
[the remark this answers is not in the document]
Comment (cbarber) [reply app/javascript/components/widget-controller.jsx]: :thinking: That will get a little tricky. I also want to have a prefetch occur before the transition to make it feel more seamless.

File: app/javascript/components/widget-controller.jsx (hunk 1 above)
Comment (cbarber) [reply app/javascript/components/widget-controller.jsx]: I'm not sure what hooks would improve here.

**Pull request merged**